api/api-common/src/test/kotlin/io/qalipsis/api/meter/specification/CounterMeterStepSpecificationTest.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * QALIPSIS
+ * Copyright (C) 2025 AERIS IT Solutions GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package io.qalipsis.api.meter.specification
+
+import io.qalipsis.api.context.StepContext
+import io.qalipsis.api.steps.CounterMeterStepSpecification
+import io.qalipsis.api.steps.counter
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+
+/**
+ * @author Francisca Eze
+ */
+internal class CounterMeterStepSpecificationTest {
+
+    @Test
+    internal fun `should add counter step without failure specification as next`() {
+        val previousStep = DummyStepSpecification()
+        val specification: (context: StepContext<Int, Int>, input: Int) -> Double =
+            { _, _ -> 12.0 }
+        previousStep.counter(name = "test-counter", specification)
+
+        assertEquals(
+            CounterMeterStepSpecification(
+                "test-counter",
+                specification
+            ), previousStep.nextSteps[0]
+        )
+    }
+
+
+    @Test
+    internal fun `should add counter step with failure specification as next`() {
+        val previousStep = DummyStepSpecification()
+        val specification: (context: StepContext<Int, Int>, input: Int) -> Double =
+            { _, _ -> 12.0 }
+        previousStep.counter(name = "test-counter", specification)
+            .shouldFailWhen {
+                count.isLessThan(12.0)
+                return@shouldFailWhen 0
```

Comment:
I'm not a fan of the return here. We should avoid it. What does the 0 mean? I rather expect:
```
 previousStep.counter(name = "test-counter", specification)
            .shouldFailWhen {
                count.isLessThan(12.0)
            }
```